src/github.com/mozilla-services/pushgo/simplepush/error.go

```diff
@@ -15,47 +15,50 @@ func (err MultipleError) Error() string {
-	ErrNoKey              ErrorCode = 110
-	ErrInvalidKey         ErrorCode = 111
-	ErrNoParams           ErrorCode = 112
-	ErrInvalidParams      ErrorCode = 113
-	ErrNoData             ErrorCode = 114
-	ErrNonexistentRecord  ErrorCode = 115
-	ErrRecordUpdateFailed ErrorCode = 116
-	ErrBadPayload         ErrorCode = 117
+// Error returns a human-readable error message.
+func (err *ServiceError) Error() string {
+	return err.Message
+}
 
-	ErrTooManyPings ErrorCode = 201
+// Status returns the HTTP status code associated with the error.
+func (err *ServiceError) Status() int {
+	return err.StatusCode
+}
 
-	ErrBadVersion  ErrorCode = 301
-	ErrDataTooLong ErrorCode = 302
+// Service errors.
+var (
+	ErrUnknownCommand     = &ServiceError{101, http.StatusUnauthorized, "Unknown command"}
+	ErrInvalidCommand     = &ServiceError{102, http.StatusUnauthorized, "Invalid Command"}
+	ErrNoID               = &ServiceError{103, http.StatusUnauthorized, "Missing device ID"}
+	ErrInvalidID          = &ServiceError{104, http.StatusServiceUnavailable, "Invalid device ID"}
+	ErrExistingID         = &ServiceError{105, http.StatusServiceUnavailable, "Device ID already assigned"}
+	ErrNoChannel          = &ServiceError{106, http.StatusUnauthorized, "No Channel ID Specified"}
+	ErrInvalidChannel     = &ServiceError{107, http.StatusServiceUnavailable, "Invalid Channel ID Specified"}
+	ErrExistingChannel    = &ServiceError{108, http.StatusServiceUnavailable, "Channel Already Exists"}
+	ErrNonexistentChannel = &ServiceError{109, http.StatusServiceUnavailable, "Nonexistent channel ID"}
+	ErrNoKey              = &ServiceError{110, http.StatusUnauthorized, "No primary key value specified"}
+	ErrInvalidKey         = &ServiceError{111, http.StatusInternalServerError, "Invalid Primary Key Value"}
+	ErrNoParams           = &ServiceError{112, http.StatusUnauthorized, "Missing required fields for command"}
+	ErrInvalidParams      = &ServiceError{113, http.StatusUnauthorized, "An Invalid value was specified"}
+	ErrNoData             = &ServiceError{114, http.StatusServiceUnavailable, "No Data to Store"}
+	ErrNonexistentRecord  = &ServiceError{115, http.StatusServiceUnavailable, "No record found"}
+	ErrRecordUpdateFailed = &ServiceError{116, http.StatusServiceUnavailable, "Error updating channel record"}
 
-	ErrServerError ErrorCode = 999
-)
+	ErrTooManyPings = &ServiceError{201, http.StatusUnauthorized, "Client sent too many pings"}
 
-// Error returns a human-readable error message.
-func (c ErrorCode) Error() string {
-	return codeToError[c].Message
-}
+	ErrBadVersion  = &ServiceError{301, http.StatusBadRequest, "Invalid update version"}
+	ErrDataTooLong = &ServiceError{302, http.StatusRequestEntityTooLarge, "Request payload too large"}
 
-// Status returns the HTTP status code associated with the error code.
-func (c ErrorCode) Status() int {
-	return codeToError[c].StatusCode
-}
+	ErrServerError = &ServiceError{999, http.StatusInternalServerError, "Internal server error"}
+)
```

Comment:
Good call. The loose convention I've been following is: 100 for invalid client input, 200 for bad client behavior, and 300 for invalid app server input (999 matches FxA). What do you think?